internal/api/mfa_test.go

```diff
@@ -516,8 +515,149 @@ func enrollAndVerify(ts *MFATestSuite, user *models.User, token string) (verifyR
 	req.Header.Set("Content-Type", "application/json")
 
 	ts.API.handler.ServeHTTP(y, req)
-	require.Equal(ts.T(), http.StatusOK, y.Code)
+	if guaranteeSuccess {
+		require.Equal(ts.T(), http.StatusOK, y.Code)
+	}
 	verifyResp = &AccessTokenResponse{}
 	require.NoError(ts.T(), json.NewDecoder(y.Body).Decode(&verifyResp))
 	return verifyResp
 }
+
+func (ts *MFATestSuite) TestVerificationHookDefaultSuccess() {
```

Comment:
All four tests here have repetition and can probably be condensed into one test iterating over the various cases